components/style/properties.mako.rs

```diff
@@ -128,38 +143,55 @@ class StyleStruct(object):
         self.gecko_ffi_name = gecko_ffi_name
         self.additional_methods = additional_methods or []
 
-STYLE_STRUCTS = []
-THIS_STYLE_STRUCT = None
-LONGHANDS = []
-LONGHANDS_BY_NAME = {}
-DERIVED_LONGHANDS = {}
-SHORTHANDS = []
-CONFIG = {}
-
-def set_product(p):
-    global CONFIG
-    CONFIG['product'] = p
+%>
 
-def new_style_struct(name, is_inherited, gecko_name=None, additional_methods=None):
-    global THIS_STYLE_STRUCT
+<%def name="new_style_struct(name, is_inherited, gecko_name=None, additional_methods=None)">
```

Comment:
Very much the key to getting stuff working in the `%include`d files is to move all stuff that needs to be callable into proper `%def` methods.